include/alpaka/kernel/TaskKernelGenericSycl.hpp

```diff
@@ -0,0 +1,305 @@
+                : WorkDivMembers<TDim, TIdx>(std::forward<TWorkDiv>(workDiv))
+                , m_kernelFnObj{kernelFnObj}
+                , m_args{args...}
+            {
+                static_assert(
+                    Dim<typename std::decay<TWorkDiv>::type>::value == TDim::value,
+                    "The work division and the execution task have to be of the same dimensionality!");
+            }
+
+            auto operator()(sycl::handler& cgh) -> void // Don't remove the trailing void or DPCPP will complain.
+            {
+                // wait for previous kernels to complete
+                cgh.depends_on(pimpl->dependencies);
+
+                // Assign placeholder accessors to this command group
+                detail::require(cgh, m_args);
+
+                // transform arguments to device tuple so we can copy the arguments into device code. We can't use
+                // std::tuple for this step because it isn't standard layout and thus prohibited to be copied into SYCL
+                // device code.
+                auto device_args = detail::make_device_args(m_args, std::make_index_sequence<sizeof...(TArgs)>{});
+
+                const auto work_groups = WorkDivMembers<TDim, TIdx>::m_gridBlockExtent;
+                const auto group_items = WorkDivMembers<TDim, TIdx>::m_blockThreadExtent;
+                const auto item_elements = WorkDivMembers<TDim, TIdx>::m_threadElemExtent;
+
+                const auto global_size = get_global_size(work_groups, group_items);
+                const auto local_size = get_local_size(group_items);
+
+                // allocate shared memory -- needs at least 1 byte to make XRT happy
+                const auto shared_mem_bytes = std::max(
+                    1ul,
+                    std::apply(
+                        [&](auto const&... args) {
+                            return getBlockSharedMemDynSizeBytes<TAcc>(
+                                m_kernelFnObj,
+                                group_items,
+                                item_elements,
+                                args...);
+                        },
+                        m_args));
+
+                auto shared_accessor
+                    = sycl::accessor<std::byte, 1, sycl::access_mode::read_write, sycl::target::local>{
+                        sycl::range<1>{shared_mem_bytes},
+                        cgh};
+                auto global_fence_dummy = pimpl->fence_dummy.get_access(cgh);
+                auto local_fence_dummy = sycl::accessor<int, 1, sycl::access_mode::read_write, sycl::target::local>{
+                    sycl::range<1>{1},
+                    cgh};
+
+                // copy-by-value so we don't access 'this' on the device
+                auto k_func = m_kernelFnObj;
+
+#    ifdef ALPAKA_SYCL_STREAM_ENABLED
+                // set up device-side printing
+                constexpr auto buf_size = std::size_t{65535}; // 64 KiB for the output buffer
```

Comment:
Does the buffer size need to be configurable?

Reply:
I need to pass it some size to work. And this was the size that was supported on all tested devices (larger ones crashed).

Reply:
Ok! Maybe add your explanation as a comment.